lib/companies/schema/pending_change.ex

```diff
@@ -15,17 +15,25 @@ defmodule Companies.Schema.PendingChange do
     field :original, :map, virtual: true
 
     belongs_to :user, User
+    belongs_to :reviewer, User
 
     timestamps()
   end
 
   @doc false
-  def changeset(user, attrs) do
-    user
-    |> cast(attrs, [:action, :approved, :changes, :note, :resource, :user_id])
+  def changeset(pending_change, attrs) do
+    pending_change
+    |> cast(attrs, [:action, :changes, :resource, :user_id])
     |> validate_required([:action, :changes, :resource, :user_id])
     |> validate_inclusion(:action, ["create", "update", "delete"])
     |> validate_inclusion(:resource, ["company", "industry", "job"])
     |> assoc_constraint(:user)
   end
+
+  def approve_changeset(pending_change, attrs) do
```

Comment:
I split up the changeset functions to one for creating the pending change, and another for approving it.